src/nbl/asset/interchange/CIESProfileLoader.cpp

```diff
@@ -253,105 +11,26 @@ CIESProfileLoader::loadAsset(io::IReadFile* _file,
-        pos.y = core::sign(pos.y) * (1.0 - abs_x);
     }
 
-    return core::normalize(pos);
-}
+    auto meta = core::make_smart_refctd_ptr<CIESProfileMetadata>(profile);
+    core::smart_refctd_ptr<asset::ICPUImageView> cpuImageView;
 
-static inline std::pair<float,float> sphericalDirToAngles(core::vectorSIMDf dir) {
-    float theta = std::acos(dir.z);
-    float phi = std::atan2(dir.y, dir.x);
-    return { theta, phi < 0 ? phi + 2 * core::PI<float>() : phi };
-}
-
-core::smart_refctd_ptr<asset::ICPUImage>
-CIESProfileLoader::createTexture(const CIESProfile& profile, size_t width, size_t height) {
-    asset::ICPUImage::SCreationParams imgInfo;
-    imgInfo.type = asset::ICPUImage::ET_2D;
-    imgInfo.extent.width = width;
-    imgInfo.extent.height = height;
-    imgInfo.extent.depth = 1u;
-    imgInfo.mipLevels = 1u;
-    imgInfo.arrayLayers = 1u;
-    imgInfo.samples = asset::ICPUImage::ESCF_1_BIT;
-    imgInfo.flags = static_cast<asset::IImage::E_CREATE_FLAGS>(0u);
-    imgInfo.format = asset::EF_R16_UNORM;
-    auto outImg = asset::ICPUImage::create(std::move(imgInfo));
-
-    asset::ICPUImage::SBufferCopy region;
-    size_t texelBytesz = asset::getTexelOrBlockBytesize(imgInfo.format);
-    size_t bufferRowLength =
-        asset::IImageAssetHandlerBase::calcPitchInBlocks(width, texelBytesz);
-    region.bufferRowLength = bufferRowLength;
-    region.imageExtent = imgInfo.extent;
-    region.imageSubresource.baseArrayLayer = 0u;
-    region.imageSubresource.layerCount = 1u;
-    region.imageSubresource.mipLevel = 0u;
-    region.bufferImageHeight = 0u;
-    region.bufferOffset = 0u;
-    auto buffer = core::make_smart_refctd_ptr<asset::ICPUBuffer>(
-        texelBytesz * bufferRowLength * height);
-
-    double maxValue = profile.getMaxValue();
-    double maxValueRecip = 1.0 / maxValue;
-
-    double vertInv = 1.0 / height;
-    double horiInv = 1.0 / width;
-    char* bufferPtr = reinterpret_cast<char*>(buffer->getPointer());
-    for (size_t i = 0; i < height; i++) {
-        for (size_t j = 0; j < width; j++) {
-            auto dir = octahdronUVToDir(j * vertInv, i * horiInv);
-            auto [theta, phi] = sphericalDirToAngles(dir);
-            double I = profile.sample(core::degrees(theta), core::degrees(phi));
-            uint16_t value = static_cast<uint16_t>(
-                std::clamp(I * maxValueRecip * 65535.0, 0.0, 65535.0));
-            *reinterpret_cast<uint16_t*>(
-                bufferPtr + i * bufferRowLength * texelBytesz + j * texelBytesz) =
-                value;
-        }
-    }
+    if (_params.loaderFlags & E_LOADER_PARAMETER_FLAGS::ELPF_LOAD_METADATA_ONLY)
+        cpuImageView = _override->findDefaultAsset<ICPUImageView>("nbl/builtin/image_view/dummy2d", loadContex, _hierarchyLevel).first; // note: we could also pass empty content, but this would require adjusting IAssetLoader source to not attempt to use all loaders to find the asset
+    else
+        cpuImageView = profile.createIESTexture();
```

Comment:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/630#discussion_r1431351882

Reply:
going in future PR